dts/bindings/pwm/ti,mspm0-pwm.yaml

```diff
@@ -0,0 +1,60 @@
+# Copyright (C) 2024 Bang & Olufsen A/S, Denmark
```

Comment:
The bindings here look significantly different than most of the other pwm drivers. My gut is telling me that we're holding it wrong in that case.

More specifically I'd expect the pwm node to look more like:
```
pwm0: pwm@40860000 {
        compatible = "ti,mspm0-pwm";
        reg = <0x40860000 0x2000>;
        ti,clock-sel = "bus"; // choice of "busclk", "2xbusclk", "mfclk", "lfclk"
        ti,prescaler = <16>; // the prescaler + divider could be omitted or collected as a requested frequency instead
        ti,divider = <16>; // see comment above
        ti,mode = "center-align";
        #pwm-cells = <2>; // where cells are <cc_index period>
};
```
Usage of a pwm is then done by
```
blinky@0 {
        compatible = "blinky,pwm";
        pwm = <&pwm0 2 100000>; // get "pwm0" with "cc_index: 2" and "period: 1000000" (assumed microseconds gives 10 Hz)
        ... 
};
```
but to be honest, I did not look that deep into the pwm bindings. It just seems more like how the rest do - potentially even omitting the prescaler, divider, and mode properties. But since it is crucial for resolution control, to be able to choose the base clock frequency (via. prescale + divider) then it makes sense to keep them either individually or combined as a single `frequency` property used to calculate them both. Considering that two of the timers do not actually support a prescaler, it might make more sense to leverage the divider only when calculating the necessary divider to hit the requested frequency.

Reply:
Fair point, did a big round and purged a lot of stuff you mentioned. Now i try to deduce each property based on the provided hw timer address (mspm0 chips use the same address for the TIMx regs)